### ISSUE TITLE
Fails in dev environment @ createSchemaCustomization

### ISSUE THREAD

**Issue by RobinSJames** [gatsbyjs/gatsby#29535]
For the past few days this issue has been preventing me from making changes to our code. I will not use the code from my repo as it is private but I have replicated this via the [Gatsby WordPress tutorial](https://github.com/gatsbyjs/gatsby-source-wordpress-experimental/blob/master/docs/tutorials/building-a-new-site-wordpress-and-gatsby.md) and encountered the exact same problem.

Specification of plugins:

- WP Gatsby 0.9.1
- WP GraphQL 1.1.1
- gatsby-source-wordpress-experimental 6.2.0
- WP 5.6

Once I've updated gatsby-config.js I run `gatsby develop` and encounter the following issue:

```success open and validate gatsby-configs - 0.033s
warn Warning: there are unknown plugin options for "gatsby-source-wordpress-experimental": protocol,
restApiRoutePrefix, hostingWPCOM, useACF
success load plugins - 0.871s
success onPreInit - 0.034s
success initialize cache - 0.019s
success copy gatsby files - 0.062s
success onPreBootstrap - 0.017s
success  gatsby-source-wordpress  ensuring plugin requirements are met - 2.294s
⠀
info  gatsby-source-wordpress

	This is either your first build or the cache was cleared.
	Please wait while your WordPress data is synced to your Gatsby cache.

	Maybe now's a good time to get up and stretch? :D

success  gatsby-source-wordpress  ingest WPGraphQL schema - 2.367s

 ERROR

There was an error

not finished createSchemaCustomization - 4.703s
```

As you can see the error provides no feedback on what could be causing this problem.
Please can you help me address this issue?

**Comment by lwz7512** [comment]
The same error: `not finished createSchemaCustomization -` occurred with the latest plugins.

my configuration:
- gatsby-source-wordpress: ^4.0.0 => 4.0.1
- WP Gatsby version 1.0.1
- WP GraphQL version 1.1.3

The `gatsby-source-wordpress` options.url default value `https://wpgatsbydemo.wpengine.com/graphql`  works well with above config, but when I changed it to my website graphql api: `https://www.wp4speed.com/graphql`, building process failed.

I'm curious about the 2 plugins(WP Gatsby, WP GraphQL) version in `wpgatsbydemo.wpengine.com`, anyone know this? @jacksellwood

**Comment by lwz7512** [comment]
I figured it out!

There is a null value of `transformedFields` in [build-types.js](https://github.com/gatsbyjs/gatsby/blob/862637be84622d945ec21c72b208c0a657c0f130/packages/gatsby-source-wordpress/src/steps/create-schema-customization/build-types.js) on line 128. So, to fix this bug, null check is required:

```
  if (!transformedFields || !Object.keys(transformedFields).length) {
    return;
  }

```

**Comment by lwz7512** [comment]
A workaround for this issue temporarily is using my [patched starter](https://github.com/lwz7512/gatsby-starter-wordpress-blog):

```
% gatsby new my-gatsby-blog-site https://github.com/lwz7512/gatsby-starter-wordpress-blog
```

This starter is a fork version of official Wordpress blog starter which includes a patch script `fix-gsw.js`:

```
"postinstall": "node ./fix-gsw.js",
```

Enjoy!

**Comment by TylerBarnes** [comment]
This should be fixed as of `gatsby-source-wordpress@4.0.2` - thanks @lwz7512 !

**Comment by lwz7512** [comment]
Awesome! @TylerBarnes  love to see this!
